### PR TITLE
Feature: Add support for Access Control Lists

### PR DESCRIPTION
**Pull request by bhudgens**
Per User access control:

  - Add a file to your service directory with [*.acl] extension:  users.acl
  - populate file with usernames (one per line)

Users that do not exist in the ACL file will be rewritten to:  /autherror

### CODE REVIEW

File: scripts/starphleet-nginx-acl-rules [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+###
+
+die_on_error
+run_as_root_or_die
+
+#******************************************************************
+# Configuration
+#******************************************************************
+
+NGINX_CONF=${NGINX_CONF:-/var/starphleet/nginx}
```
Comment (igroff) [inline scripts/starphleet-nginx-acl-rules]: NGINX_CONF is already guaranteed by the /etc/starphleet file, and pulled in via tools via starphleet-launcher adding a default here complicates things and is just noise

File: scripts/starphleet-nginx-acl-rules [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+DIR_NGINX_ACL_RULES="${NGINX_CONF}/acl_rules"
+DIR_NGINX_PUBLISHED="${NGINX_CONF}/published"
+
+#******************************************************************
+# Main
+#******************************************************************
+
+# We purge everything each run and start over
+[ -d "${DIR_NGINX_ACL_RULES}" ]  && rm -rf "${DIR_NGINX_ACL_RULES}"
+rm "${DIR_NGINX_PUBLISHED}"/*.acl 2> /dev/null || true
```
Comment (igroff) [inline scripts/starphleet-nginx-acl-rules]: Do not want to hide errors, if there's something unexpected.... we'll never be able to see it and will have to remove the 2> /dev/null before we can figure out what it is.

File: scripts/starphleet-nginx-acl-rules [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+# Make sure our dirs work
+mkdir -p "${DIR_NGINX_ACL_RULES}"
+mkdir -p "${DIR_NGINX_PUBLISHED}"
+
+# Go through all Access Control List files (*.acl)
+for FILE_ACL_LIST in $(find ${HEADQUARTERS_LOCAL} -type f -name "*.acl"); do
+  # Yoink the HQ Path off
+  SERVICE=$(echo ${FILE_ACL_LIST} | perl -ne "s|${HEADQUARTERS_LOCAL}/||;print $1")
```
Comment (igroff) [inline scripts/starphleet-nginx-acl-rules]: please no perl, we don't have it yet let's not add it.

File: scripts/starphleet-nginx-acl-rules [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  # Yoink the file name for theFILE_ACL_LIST
+  SERVICE=$(echo ${SERVICE} | perl -ne "s|/$(basename ${FILE_ACL_LIST})||;print $1")
```
Comment (igroff) [inline scripts/starphleet-nginx-acl-rules]: same as pervious perl comment

File: scripts/starphleet-nginx-acl-rules (hunk 2 above)
Comment (bhudgens) [inline scripts/starphleet-nginx-acl-rules]: Okay - but it's valid that there were no ACL files previously.. which generates an error.  Thus, I hid them.

File: scripts/starphleet-nginx-acl-rules [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+# Go through all Access Control List files (*.acl)
+for FILE_ORDERS in $(find ${HEADQUARTERS_LOCAL} -type f -name "orders"); do
+  # Yoink the HQ Path off
+  SERVICE=$(echo ${FILE_ORDERS#${HEADQUARTERS_LOCAL}/})
+  # Yoink the file name for theFILE_ACL_LIST
+  SERVICE=$(echo ${SERVICE} | sed -e "s/$(basename ${FILE_ORDERS})//")
+  # Yoink the trailing slash
+  SERVICE=${SERVICE::-1}
+  # The file with our rules for this service
```
Comment (igroff) [inline scripts/starphleet-nginx-acl-rules]: I think you're just trying to get the name of the directory that contains the orders file here, no?  This is very tortured, how about something like basename $(dirname ~/Development/starphleet/server/overlay/pant.acl)

**Pull request merged**